libs/api/domains/cms/src/lib/models/districts.model.ts

```diff
@@ -0,0 +1,36 @@
+import { Field, ID, ObjectType } from '@nestjs/graphql'
+
+import { IDistricts } from '../generated/contentfulTypes'
+import { Image, mapImage } from './image.model'
+import { Link, mapLink } from './link.model'
+import { SystemMetadata } from 'api-cms-domain'
+
+@ObjectType()
+export class Districts {
+  @Field(() => ID)
+  id: string
+
+  @Field()
+  title: string
+
+  @Field({ nullable: true })
+  description: string
+
+  @Field({ nullable: true })
+  image?: Image
+
+  @Field(() => [Link])
+  links?: Array<Link>
+}
+
+export const mapDistricts = ({
+  sys,
+  fields,
+}: IDistricts): SystemMetadata<Districts> => ({
+  typename: 'Districts',
+  id: sys.id,
+  title: fields.title,
+  description: fields.description,
+  image: mapImage(fields.image),
+  links: fields.links.map(mapLink),
```

Comment:
We have a way to resolve internal and external links with references using a model called `ReferenceLink`. Our goal is to consolidate all link logic into this model in future to allow for dynamic link resolution for all pages types.
If you think this fits your case you can change link to a reference field an pass it's value into `mapReferenceLink` you can use the menu link for reference the model is called `MenuLink` and the content type exists [here](https://app.contentful.com/spaces/8k0h54kbe6bj/content_types/menuLink/fields) in Contentful.
This might apply to all links in the following models